R/create_profile_matrix.R

```diff
@@ -0,0 +1,243 @@
+#'                               cellNames))
+#' cellAnnots <- as.data.frame(cbind(CellID=cellNames, cellType=sample(size = length(cellNames), replace = TRUE, 
+#'                                                                     x = c("A", "B", "C", "D"))))
+#' table(cellAnnots$cellType)
+#' profile_matrix <- create_profile_matrix(mtx = mtx,  
+#'                                         cellAnnots = cellAnnots, 
+#'                                         cellTypeCol = "cellType",
+#'                                         cellNameCol = "CellID",
+#'                                         minGenes = 10,
+#'                                         scalingFactor = 1)
+#' head(profile_matrix)
+#' @export
+
+create_profile_matrix <- function(mtx, cellAnnots, cellTypeCol, cellNameCol, matrixName = "Custom", 
+                                  outDir = "./", geneList = NULL, normalize = FALSE, scalingFactor = 5, minCellNum = 15, 
+                                  minGenes = 100, numCellTypesExpr = 1, filterCellTypes = TRUE) {
+  
+  # checking user input values
+  if(is.null(mtx)){
+    stop("count matrix is necessary")
+  }
+  if (!dir.exists(outDir)) {
+    stop("Output directory is not valid")
+  }
+  if (is.null(cellAnnots)){
+    stop("Cell Annotations are needed")
+  }
+    
+  if(is.null(cellTypeCol)){
+    stop("cellTypeCol must not be NULL")
+  }else if (!cellTypeCol %in% colnames(cellAnnots)){
+    stop("cellTypeCol not in cellAnnots")
+  }
+    
+  if(is.null(cellNameCol)){
+    stop("cellNameCol must not be NULL")
+  }else if (!cellNameCol %in% colnames(cellAnnots)){
+    stop("cellNameCol not in cellAnnots")
+  }
+  
+  if(class(normalize) != "logical"){
+    warning("normalize not a boolean, continuing with assumption that data should not be normalized")
+    normalize <- TRUE
+  }
+  if(class(filterCellTypes) != "logical"){
+    warning("filterCellTypes not a boolean, continuing with default of filtering cell types")
+    filterCellTypes <- TRUE
+  }
+  
+  if(class(scalingFactor) != "numeric"){
+    warning("scalingFactor not a numeric, continuing with default value of 5")
+    scalingFactor <- 5
+  }
+  if(class(minCellNum) != "numeric"){
+    warning("minCellNum not a numeric, continuing with default value of 15")
+    minCellNum <- 15
+  }
+  if(class(minGenes) != "numeric"){
+    warning("minGenes not a numeric, continuing with default value of 100")
+    minGenes <- 100
+  }
+  if(class(numCellTypesExpr) != "numeric"){
+    warning("numCellTypesExpr not a numeric, continuing with default value of 1")
+    numCellTypesExpr <- 1
+  }
+  
+  #make a sparse matrix
+  mtx<- Matrix::Matrix(as.matrix(mtx), sparse = T) 
+  
+  cellTypes <- NULL
+  
+  #read in cell type annotation file
+  #get cell types 
+  cellTypes <- cellAnnots[[cellTypeCol]]
+  #assign cell name to type
+  names(cellTypes) <- cellAnnots[[cellNameCol]]
+  
+  if(is.null(cellTypes)){
+    stop("Individual cell's type are not valid")
```

Comment:
Maybe instead write, "cellAnnots and/or cellTypeCol arguments are incorrectly formatted. cellAnnots should be a data frame, and cellTypeCol should give the name of the column holding each cell's cell type."